### ISSUE TITLE
fix release pyiceberg-core to pypi

### ISSUE THREAD

**Issue by kevinjqliu** [apache/iceberg-rust#2361]
We made python publishing a "reusable workflow" and [part of publish.yml](https://github.com/apache/iceberg-rust/blob/4b0b35255469039d9877218049f93b07556124b5/.github/workflows/publish.yml#L72-L82)
Apparently this isnt a supported path for PyPi trusted publishing... 

Ran into this in 0.9.1 release. Workaround is to run release_python.yml directly

```
##[debug]Authenticating to https://upload.pypi.org/legacy/ via Trusted Publishing
##[debug]Selected Trusted Publishing Exchange Endpoint: https://upload.pypi.org/_/Oidc/Mint-Token
Warning: Potential workflow misconfiguration: 
The claims in this token suggest that the calling workflow is a reusable workflow.

In particular, this action was initiated by:

    apache/iceberg-rust/.github/workflows/release_python.yml@refs/tags/v0.9.1-rc.2

Whereas its parent workflow is:

    apache/iceberg-rust/.github/workflows/publish.yml@refs/tags/v0.9.1-rc.2

Reusable workflows are **not currently supported** by PyPI's Trusted Publishing
functionality, and are subject to breakage. Users are **strongly encouraged**
to avoid using reusable workflows for Trusted Publishing until support
becomes official. Please, do not report bugs if this breaks.
```

**Comment by dannycjones** [comment]
We just ran into this on 0.10 also. I have some ideas on how to solve it, but for v0.10.0-rc.1 we've followed the workaround (https://github.com/apache/iceberg-rust/actions/runs/28161095313). cc @CTTY

**Comment by dannycjones** [comment]
I'm totally new to the Python bindings, but here's my current thoughts:

- To fix the full release process, I'm considering the pull PyPI publishing out of the current reusable workflow for prod PyPI release, and put it in `publish.yml` (which gets us around the blocker on trusted publishing and reusable workflows).
- There's a bit of duplication with the release and the nightly publishing of Python bindings. Given the move above already to make the workflow focus on package building only, we might consider having a reusable workflow for that alone and simply delegate. We might parameterize the workflow a bit, such that we say "this is nightly" or "this is an official release" and branch the matrix generation, etc. on that. Then `publish.yml` and `release_python_nightly.yml` can become thin wrappers on the bulk of the Python packaging workflows.

I have proposed a fix for the first point for now - pulling PyPI publishing into `publish.yml` and unblocking releases: https://github.com/apache/iceberg-rust/pull/2715. However, I had tried to understand how we might better factor the workflows and figured I'd share my thinking if anyone is interested in picking that up.
